### PR TITLE
[camera_android] Set buildconfig to true for compatibility with AGP 8.0+

### PR DESCRIPTION
**Pull request by gmackall**
Fixes https://github.com/flutter/flutter/issues/134818

See https://github.com/flutter/packages/pull/4535 for an example of when this was done for a different plugin.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by stuartmorgan** [comment]
test-exempt: We can't test this until we have the right version of Java in CI; the plan is to look into rolling all our examples forward to 8 once we migrate Android tests to LUCI, since it should hopefully be easier to manage our Java version at that point.

**Comment by stuartmorgan** [comment]
test-exempt: build configuration change is a test

File: packages/camera/camera_android/android/build.gradle [hunk 1]
```diff
@@ -9,7 +9,7 @@ buildscript {
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:7.0.2'
+        classpath 'com.android.tools.build:gradle:7.3.0'
```
Comment (camsim99) [inline packages/camera/camera_android/android/build.gradle:12]: Can you bump this version too or do you run into the same issues as https://github.com/flutter/packages/pull/4597?

Comment (gmackall) [reply packages/camera/camera_android/android/build.gradle]: Trying it now

Comment (gmackall) [reply packages/camera/camera_android/android/build.gradle]: I also tried upgrading **only** the plugin AGP version (not the example) and that didn't cause any issues, so it should be fine to add this upgrade to this pr

**Review by camsim99 — APPROVED** [review]
LGTM! We likely will need to replicate these steps for the rest of the dependabot updates to AGP 8+.

**Comment by stuartmorgan** [comment]
> LGTM! We likely will need to replicate these steps for the rest of the dependabot updates to AGP 8+.

I think `camera` and `url_launcher` are the only plugins where we are currently using `BuildConfig`.

**Pull request merged**